crates/api/src/arc.rs

```diff
@@ -0,0 +1,386 @@
+//! Kitsune2 arcs represent a range of hash locations on the DHT.
+//!
+//! An arc can exist in its own right and refer to a range of locations on the DHT.
+//! It can also be used in context, such as an agent. Where it represents the range of locations
+//! that agent is responsible for.
+
+use serde::{Deserialize, Serialize};
+
+/// A basic definition of a storage arc compatible with the concept of
+/// storage and querying of items in a store that fall within that arc.
+#[derive(Debug, Clone, Serialize, Deserialize, Eq, PartialEq, Default)]
+pub enum BasicArc {
+    /// No DHT locations are contained within this arc.
+    #[default]
+    Empty,
+    /// All DHT locations are contained within this arc.
+    Full,
```

Comment:
This is a duplicate state. It can be represented by the Literal variant. I'd be inclined to get rid of Full, even though it makes some comparison logic simpler, just to avoid multiple variants meaning the same thing.

Reply:
It can be, and I didn't like that `Full` didn't have a canonical representation but I've gone back to inclusive bounds, so there is a single representation of full with `(0, u32::MAX)`